types/tests/model-test.ts

```diff
@@ -0,0 +1,35 @@
+import { Model, Registry } from "miragejs";
+import Schema from "miragejs/orm/schema";
+
+const PersonModel = Model.extend({
+  name: "hello"
+});
+
+declare const schema: Schema<Registry<{ person: typeof PersonModel }, {}>>;
+
+const people = schema.all("person");
+
+people.length; // $ExpectType number
+people.modelName; // $ExpectType string
+people.models.map(model => {
+  model.id; // $ExpectType string | undefined
+  model.name; // $ExpectType string
+  model.modelName; // $ExpectType string
+  model.attrs; // $ExpectType Record<string, unknown>
+  model.foo; // $ExpectError
+
+  model.save();
+  model.reload();
+  model.destroy();
```

Comment:
We should assert that these are all `void` IIRC.

Reply:
Assertions added.